### PR TITLE
[SPARK-38742][SQL][TESTS] Move the tests `MISSING_COLUMN` from SQLQuerySuite to QueryCompilationErrorsSuite

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This pr aims to move tests for the error class MISSING_COLUMN from SQLQuerySuite to QueryCompilationErrorsSuite, it's a followup of SPARK-37935.

### Why are the changes needed?
To improve code maintenance.
 
### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
By running the moved tests:
```
$ build/sbt "sql/testOnly *QueryCompilationErrorsSuite*"
```

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala [hunk 1]
```diff
@@ -362,6 +362,45 @@ class QueryCompilationErrorsSuite extends QueryTest with SharedSparkSession {
...
+
+  test("MISSING_COLUMN: SELECT distinct does not work correctly if order by missing attribute") {
+    checkAnswer(
+      sql("""select distinct struct.a, struct.b
+            |from (
+            |  select named_struct('a', 1, 'b', 2, 'c', 3) as struct
+            |  union all
+            |  select named_struct('a', 1, 'b', 2, 'c', 4) as struct) tmp
+            |order by a, b
+            |""".stripMargin),
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala:374]: Fix indentation here.

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala [hunk 2]
```diff
@@ -362,6 +362,45 @@ class QueryCompilationErrorsSuite extends QueryTest with SharedSparkSession {
...
+            |from (
+            |  select named_struct('a', 1, 'b', 2, 'c', 3) as struct
+            |  union all
+            |  select named_struct('a', 1, 'b', 2, 'c', 4) as struct) tmp
+            |order by struct.a, struct.b
+            |""".stripMargin)
+    }
+    assert(e.getErrorClass === "MISSING_COLUMN")
+    assert(e.message === "Column 'struct.a' does not exist. " +
+      "Did you mean one of the following? [a, b]")
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala:388]: Use `checkErrorClass()`, please.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Comment by MaxGekk** [comment]
@panbingkun Could you resolve conflicts, please.

**Comment by panbingkun** [comment]
> @panbingkun Could you resolve conflicts, please.

Waiting for https://github.com/apache/spark/pull/36298
They modify the same file.

**Comment by panbingkun** [comment]
> @panbingkun Could you resolve conflicts, please.

Done

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala [hunk 3]
```diff
@@ -409,6 +409,69 @@ class QueryCompilationErrorsSuite
...
+            |from (
+            |  select named_struct('a', 1, 'b', 2, 'c', 3) as struct
+            |  union all
+            |  select named_struct('a', 1, 'b', 2, 'c', 4) as struct) tmp
+            |order by struct.a, struct.b
+            |""".stripMargin)
+      },
+      errorClass = "MISSING_COLUMN",
+      msg = s"""Column 'struct.a' does not exist. """ +
+        s"""Did you mean one of the following\\? \\[a, b\\]; line 6 pos 9;
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala:438]: Do you really need s"? I don't see any string interpolation here.

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala [hunk 4]
```diff
@@ -409,6 +409,69 @@ class QueryCompilationErrorsSuite
...
+  test("MISSING_COLUMN - SPARK-21335: support un-aliased subquery") {
+    withTempView("v") {
+      Seq(1 -> "a").toDF("i", "j").createOrReplaceTempView("v")
+      checkAnswer(sql("SELECT i from (SELECT i FROM v)"), Row(1))
+
+      checkErrorClass(
+        exception = intercept[AnalysisException](sql("SELECT v.i from (SELECT i FROM v)")),
+        errorClass = "MISSING_COLUMN",
+        msg = s"""Column 'v.i' does not exist. Did you mean one of the following\\? """ +
+          s"""\\[__auto_generated_subquery_name.i\\]; line 1 pos 7;
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala:461]: Is s""" needed here?

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**